### PR TITLE
Admin datasets

### PR DESCRIPTION
**Pull request by douglas1850**
*Description of changes:* Admins now have a page where they can view *all* datasets. We've discussed it as a team and agreed that admins should be able to view/edit/delete datasets that they are not the owners of (private) or members of the project for. This will also ensure a dataset (especially Group datasets) can never be orphaned. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: backend/src/ml_space_lambda/data_access_objects/dataset.py [hunk 1]
```diff
@@ -139,3 +139,7 @@ def get_all_for_scope(self, dataset_type: DatasetType, scope: str) -> List[Datas
         ).records
 
         return [DatasetModel.from_dict(entry) for entry in json_response]
+
+    def get_all(self) -> List[DatasetModel]:
```
Comment (estohlmann) [inline backend/src/ml_space_lambda/data_access_objects/dataset.py:143]: Probably should write a quick unit test for this

File: backend/src/ml_space_lambda/authorizer/lambda_function.py [hunk 2]
```diff
@@ -317,37 +317,33 @@ def lambda_handler(event, context):
                 if "x-mlspace-dataset-type" in event["headers"] and "x-mlspace-dataset-scope" in event["headers"]:
                     target_type = event["headers"]["x-mlspace-dataset-type"]
                     target_scope = event["headers"]["x-mlspace-dataset-scope"]
-                    if target_type == DatasetType.GLOBAL:
+                    if Permission.ADMIN in user.permissions:
```
Comment (estohlmann) [inline backend/src/ml_space_lambda/authorizer/lambda_function.py:320]: We will want to get Dustin's changes pulled in here so they are accounted for when checking permissions like this

File: backend/src/ml_space_lambda/dataset/lambda_functions.py [hunk 3]
```diff
@@ -305,23 +305,32 @@ def create_dataset(event, context):
 @api_wrapper
 def list_resources(event, context):
     username = event["requestContext"]["authorizer"]["principalId"]
+    is_admin = event["requestContext"].get("resourcePath") == "/admin/datasets"
```
Comment (dustins) [inline backend/src/ml_space_lambda/dataset/lambda_functions.py:308]: This is perfectly valid but it feels a little something to be basing if they're an admin on the URL. Especially with how hard to follow our authorizer is. Not blocking but I'd suggest this be changed to closer to how it is done [in a few other places](https://github.com/awslabs/mlspace/blob/develop/backend/src/ml_space_lambda/group/lambda_functions.py#L67-L69).

```
    user = UserModel.from_dict(json.loads(event["requestContext"]["authorizer"]["user"]))
    if Permission.ADMIN in user.permissions:
```

File: frontend/src/entities/dataset/detail/dataset-detail.tsx [hunk 4]
```diff
@@ -55,7 +62,7 @@ function DatasetDetail () {
     datasetDetails.set('Access level', showAccessLevel(dataset));
     datasetDetails.set('Location', dataset.location);
     if (dataset.type === DatasetType.GROUP) {
-        datasetDetails.set(`Group${dataset.groups && dataset.groups.length > 1 ? 's' : ''}`, dataset.groups);
+        datasetDetails.set(`Group${dataset.groups && dataset.groups.length > 1 ? 's' : ''}`, dataset.groups?.length === 0 ? 'None' : dataset.groups);
```
Comment (dustins) [inline frontend/src/entities/dataset/detail/dataset-detail.tsx:65]: I'm not sure if this actually occurs but it looks like `dataset.groups?.length` could return undefined. In which case it isn't `=== 0` and would try and display `dataset.groups`. Might be better to let truthiness rule the day…

```typescript
dataset.groups?.length ? dataset.groups : 'None'
```

File: frontend/src/entities/dataset/dataset.tsx [hunk 5]
```diff
@@ -65,14 +87,14 @@ export const Dataset = () => {
...
-                    visibleColumns={visibleColumns}
+                    columnDefinitions={isAdmin ? adminDatasetColumns : defaultColumns }
+                    visibleColumns={isAdmin ? visibleAdminColumns : visibleColumns}
                     visibleContentPreference={visibleContentPreference}
                     trackBy='location'
                     itemNameProperty='name'
                     loadingItems={loadingDatasets}
                     loadingAction={loadingAction}
-                    serverFetch={getDatasetsList}
+                    //serverFetch={isAdmin ? getAllDatasets : getDatasetsList}
```
Comment (dustins) [inline frontend/src/entities/dataset/dataset.tsx:97]: Should this be commented out or just not exist?

File: backend/src/ml_space_lambda/dataset/lambda_functions.py (hunk 3 above)
Comment (douglas1850) [reply backend/src/ml_space_lambda/dataset/lambda_functions.py:308]: This is a good point; I think I'll combine the two. I don't want to return all datasets every time an admin tries to list datasets, only when they're listing datasets from the admin datasets page (which uses the /admin/datasets endpoint)

File: frontend/src/entities/dataset/dataset.tsx (hunk 5 above)
Comment (douglas1850) [reply frontend/src/entities/dataset/dataset.tsx:97]: I meant to talk about this in dev sync, I'm trying to get serverFetch to work with `getAllDatasets` or `getDatasetsList` but it's erroring because of typing. Wanted to see if you or Evan knew a way to get this prop to be happy with either function

File: frontend/src/entities/dataset/detail/dataset-detail.tsx (hunk 4 above)
Comment (douglas1850) [reply frontend/src/entities/dataset/detail/dataset-detail.tsx:65]: good point, will do

File: lib/config.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (douglas1850) [inline lib/config.json:1]: ah damnit must've committed this on mistake, will remove on next commit

File: backend/src/ml_space_lambda/authorizer/lambda_function.py [hunk 7]
```diff
@@ -426,13 +432,13 @@ def _handle_dataset_request(request_method, path_params, user):
     dataset_name = path_params["datasetName"]
     dataset = dataset_dao.get(dataset_scope, dataset_name)
     if dataset:
-        # Owners can do whatever - this check also handles private datasets
-        if dataset.created_by == user.username:
+        # Owners and Admins can do whatever - this check also handles private datasets
+        if dataset.created_by == user.username or Permission.ADMIN in user.permissions:
```
Comment (estohlmann) [inline backend/src/ml_space_lambda/authorizer/lambda_function.py:436]: nit: this could be simplified to `or  IS_ADMIN`

**Review by dustins — APPROVED** [review]

**Review by estohlmann — APPROVED** [review]
